addons/point_of_sale/static/src/app/screens/partner_list/partner_editor/partner_editor.js

```diff
@@ -48,13 +58,14 @@ export class PartnerDetailsEdit extends Component {
             return false;
         }
     }
-    /**
-     * Save to field `changes` all input changes from the form fields.
-     */
-    captureChange(event) {
-        this.changes[event.target.name] = event.target.value;
-    }
     saveChanges() {
+        if (
+            this.changes.state_id &&
+            this.pos.states.find((state) => state.id === parseInt(this.changes.state_id))
+                .country_id[0] !== parseInt(this.changes.country_id)
+        ) {
+            this.changes.state_id = false;
+        }
```

Comment:
```suggestion
        const stateId = parseInt(this.changes.state_id);
        const countryId = parseInt(this.change.country_id);
        if (
            this.changes.state_id &&
            this.pos.states.find((state) => state.id === stateId)
                .country_id[0] !== countryId
        ) {
            this.changes.state_id = false;
        }
```

Reply:
what do you think about this? 
`        this.changes.country_id = this.changes.country_id && parseInt(this.changes.country_id);
        this.changes.state_id = this.changes.state_id && parseInt(this.changes.state_id);
`
this way we send down the line the correct variable type

Reply:
https://github.com/odoo/odoo/pull/126021#discussion_r1243895257